models/classes/runner/QtiRunnerService.php

```diff
@@ -1097,7 +1098,20 @@ public function timeout(RunnerServiceContext $context, $scope, $ref)
             }
             try {
                 $session->closeTimer($ref, $scope);
-                $session->checkTimeLimits(false, true, false);
+                if ($late) {
```

Comment:
I am wondering how difficult it is to get the timeConstraint from the scope and ref to know if the timer has the parameter `allowLateSubmission`. If possible it would be better than relying on a parameter

Reply:
Yes, it could be possible, but it will come with some cost. And we cannot get rid of the parameter, as we we cannot simply check for the `allowLateSubmission` since the processing should not be done on each call but only when the `timeout` is actually done (i.e. when the TT submit the responses)